### ISSUE TITLE
Email signups not being added to PauseAI chat

### ISSUE THREAD

**Issue by joepio** [Maximophone/ProjectBot#10]
New members who join aren't appearing in our `onboarding-pipeline`. This problem started somewhere after august 8th.

The discord bot seems to be working properly otherwise.

**Comment by anthonybailey** [comment]
Noted. Sounds bad. Have some meetings booked but we should look with some urgency. Dump details of one particular case?

**Comment by anthonybailey** [comment]
Ah, many in https://discord.com/channels/1100491867675709580/1174807044990193775/1280510612212027443

Might as well copy/paste. Some associated date/times could help though?

Regular many field cases:
```
Robert Starkings    United Kingdom    Kingston upon Hull    robertstarkings7@gmail.com
Will M    United States    Alexandria, VA    willmccabe41@gmail.com
Katie    Australia    Melbourne    Katekenny625@gmail.com
Zoe Urban    United States    Denton, TX    zoemurban@gmail.com    I want to inform people about the dangers of AI and push for more regulations on AI
Satwant (Sati)    UK    St Albans    Satwantvirdee86@gmail.com    Any way I can!
Ashlee Matthews    USA        ashleematthews@gmail.com    Not sure
Joseph James Siriani III    United States of America    Westford    josephsiriani@gmail.com    Local organizing and lobbying (to the best of my ability)
alex romero    usa    ra    nh0xth0@gmail.com    d
Sophie Mills    England    Kent    18smills@thelangton.org.uk    I want to spread awareness about the dangers of AI.
Roshan Shah    United Kingdom    London    srhn592@gmail.com
Robert Starkings    United Kingdom    Kingston upon Hull    robertstarkings7@gmail.com
Avery Gilstrap    United States of America    Indianapolis    enameleburi@gmail.com    However I can
```

Long comment case
```
Matěj Jaroš    Czech Republic    Prague    matejjaros2@gmail.cz    "I want to start with translating and printing flyers and trying to get more members where I live in order to estabilish local community, which I can help to organise. I want to organise protests in future, but PauseAI has too few members at the moment where I live.
I am a writer in my schools magazine and I plan to publish a longer article there, and I could also send articles to other, bigger newspapers, but I can not garantee that they are going to publish it. I want to start AI safety group in my school with new school year.

I also started a youtube channel (the first and currently only AI safety oriented channel in czech language) and my own website (also the first and only one in the same cathegory, sadly), but both are currently almost empty and no one watches/reads it, so it might not be the right way for me."
```

Individual email case?
```
kartikayeinuk@gmail.com
```

**Comment by anthonybailey** [comment]
The evidence/explanation on testing in https://github.com/PauseAI/ProjectBot/pull/7 is not great! But unless there's some mistake then given it was merged Jul 5 and Joep reports a break after Aug 8 (I assume UTC, and based on some successful onboarding via this mechanism some time on the 9th?) I guess it can't be solely responsible. I will try to read the bot code better but I _imagine_ the breaking event was a change to roles in the Discord server.

**Comment by anthonybailey** [comment]
OK. So the mechanism Maxime implemented (post refactor 5 months back) looks a little arcane, but new code always does, and he has documented it well in https://github.com/PauseAI/ProjectBot/blob/main/OnboardingConfig.md

**Comment by anthonybailey** [comment]
I assume entry to this flow is signing up via [Airtable form](https://airtable.com/embed/appWPTGqZmUcs3NWu/pagoxRuCai4OYJEHt/form) linked on https://github.com/PauseAI/pauseai-website/blob/main/src/posts/join.md), and that on form submit this creates an entry in the relevant Airtable table which this bot project can see and act on, posting in the Discord channel and then some onboarder reaches out to the new member at the email provided. Not good at Airtable app conventions but seems logical.

**Comment by anthonybailey** [comment]
Dumping Will's notes from Discord.

 Studying the codebase and Discord settings, with some help from ChatGPT.  I am going to summarize my key findings here, both to keep track of it all myself and also to keep you in the loop.
- Lots of info regarding server permissions can be found by right clicking the PauseAI server icon > server settings > roles.  From there, you can view various roles to see which members have the role and what permissions the role has.
- Relevant permissions include View Channels, Manage Channels, and Manage Roles.
- As a member of the Onboarding Team, I can Manage Roles.
- As a Moderator, I can View Channels.
- Only Admins and Technicians (I am neither of these) can Manage Channels.
- ProjectBot can View Channels, Manage Roles, and Send Messages.  These are general permissions and should be all it needs to function properly.
- It is possible that the ProjectBot has overriding permissions specific to the onboarding-pipeline channel.
- The Manage Channels role is necessary to edit ProjectBot's permissions, or to see its permissions for a specific channel (like onboarding-pipeline).
- Conclusion: the error is probably not here, but I will need to be or get the help of an admin or technician to be sure.
- Technicians are: Joep, Maxime, Pato, and dasSpieler

**Comment by anthonybailey** [comment]
(Public access for these kind of notes seems ok to me, and issue updates in private messages is an anti-pattern across my experience.)

**Comment by anthonybailey** [comment]
(Me adding this next thought in passing here rather than making a Trello card may be one too but 🙂 we almost certainly want to get our current permissions across the stack increased!)

**Comment by joepio** [comment]
@anthonybailey do you think you can fix this issue? If not, I think it's time we deprecate the bot and find another solution

**Comment by anthonybailey** [comment]
Will, please confirm/deny whether I have this correct:

For the foreseeable, every day or two Will will check the airtable for form-based sign ups. As you've kindly done, Joep, he'll manually write messages in the Discord #onboarding-pipeline. To make the process visible, he'll also note if he checked and there was nothing to do. (Ideally with some message in the same channel, but if that can't be doine without breaking the bot then elsewhere.)

Will's rationale is that he wants to fully understand the Discord bot code, and how it relates to current and future onboarding, rather than doing s quick fix, and reckons the maintenance hit in the meantime is worth it.

If Will wasn't also the lead of the Onboarding Team, or if there was other concern that onboarding was too compromised, then this is not how I believe the software team should act. Supporting software and its users well can require mitigating first and fixing more fully later. But here where Will is both dev and customer, fair enough.

**Comment by joepio** [comment]
Makes sense. This is a fine temporary solution. However, having no maintenance and multiple major open bugs (also #2 #9) does make me believe that deprecating it might be the right call.

**Comment by anthonybailey** [comment]
Yep. I mentioned this to Will when we met, in that context of other issues, and noted that ultimately a CRM might be a better common store that would subsume onboarding Airtable data.

Your own thoughts on all this would be valuable, Will. I think I've understood your stance but don't want to put further words in your mouth.

**Comment by joepio** [comment]
@Will9371
